R/estimate_cumtime.R

```diff
@@ -0,0 +1,33 @@
+
+#' @title Estimate cumtime based on global speed
+#'
+#' @description Estimate cumtime of NA values of a GPS data based on the average
+#' speed of the valid GPS rows.
+#'
+#' @param gps_data A data.table with GPS-like rows.
+#' @details This function estimates speed values using the average speed of the
+#' whole GPS data, ignoring infinite, NA, and NAN values. Values above 80km/h and
+#' below 2km/h are also ignored and their values will be updated with the average.
+#' @return A GPS data with updated values of cumtime.
+#' @export
+#' @examples
+#' poa <- read_gtfs(system.file("extdata/poa.zip", package="gtfs2gps"))
+#'
+#' poa_gps <- gtfs2gps(poa)
+#' poa_gps_new <- estimate_cumtime(poa_gps)
+estimate_cumtime <- function(gps_data){
+  gps_data[, speed := as.numeric(speed)]
+  gps_data[speed == "Inf" | is.na(speed) | is.nan(speed), speed := NA]
+  gps_data[speed > 80 | speed < 2, speed := NA] # too slow or too fast
```

Comment:
I'm wondering whether we could add the following parameters:

- `max_speed`: Defaults to 80 Km/h
- `min_speed`: Defaults to 2 Km/h

I'm not sure about this other parameter:
- `constant`: Defaults to NULL so the function uses the average speed of the whole trip. If the user passes a value, for example 15 Km/h, then the function recalculates the cumtime and departur_times considering that

Reply:
I don't think I understood well what `constant` argument does, but couldn't it be a separate function?